crates/noirc_frontend/src/hir_def/types.rs

```diff
@@ -1192,9 +1202,9 @@ impl Type {
                 }
             }
 
-            (Function(params_a, ret_a), Function(params_b, ret_b)) => {
+            (Function(params_a, ret_a, _env_a), Function(params_b, ret_b, _env_b)) => {
```

Comment:
We should still unify and subtype function environments, lets add this to the list of things for a later PR.